packages/angular-material/example/app/app.component.ts

```diff
@@ -61,61 +76,52 @@ import { map } from 'rxjs/operators';
         {{ readonly ? 'Unset' : 'Set' }} Readonly
       </button>
     </div>
-    <jsonforms-outlet></jsonforms-outlet>
+    <jsonforms
+      [data]="selectedExample.data"
+      [schema]="selectedExample.schema"
+      [uischema]="selectedExample.uischema"
+      [renderers]="renderers"
+      (dataChange)="onDataChange($event)"
+      [locale]="currentLocale"
+      [uischemas]="uischemas"
+    ></jsonforms>
   `
 })
 export class AppComponent {
-  readonly exampleData$ = of(getExamples());
+  readonly renderers = angularMaterialRenderers;
+  readonly examples = getExamples();
+  selectedExample: ExampleDescription;
   currentLocale = 'en-US';
   private readonly = false;
   data: any;
+  uischemas: { tester: UISchemaTester; uischema: UISchemaElement; }[] = [
+    { tester: itemTester, uischema: uiSchema }
+  ];
 
-  constructor(private jsonformService: JsonFormsAngularService) {
-    const examples = getExamples();
-    const selectedExample = examples[0];
-    this.jsonformService.updateCore(
-      Actions.init(
-        selectedExample.data,
-        selectedExample.schema,
-        selectedExample.uischema
-      )
-    );
-
-    this.jsonformService.$state.pipe(
-      map(state => getData(state))
-    ).subscribe(data => this.data = data);
+  constructor() {
+    this.selectedExample = this.examples[19];
   }
-  examples() {
-    return getExamples();
+
+  onDataChange(data: any) {
+    this.data = data;
   }
 
-  onChange = (ev: any) => {
-    const examples = getExamples();
-    const selectedExample = examples.find(e => e.name === ev.target.value);
-    this.jsonformService.updateCore(
-      Actions.init(
-        selectedExample.data,
-        selectedExample.schema,
-        selectedExample.uischema
-      )
-      );
-      this.jsonformService.updateConfig(setConfig(selectedExample.config));
-      this.jsonformService.updateLocale(setLocale(this.currentLocale));
-  };
+  onChange(ev: any) {
+    this.selectedExample = this.examples.find(e => e.name === ev.target.value);
+  }
 
   changeLocale(locale: string) {
     this.currentLocale = locale;
-    this.jsonformService.updateLocale(setLocale(locale));
   }
 
   setReadonly() {
-    const uischema = getUiSchema(this.jsonformService.getState());
+    const newUiSchema = clone(this.selectedExample.uischema);
     if (this.readonly) {
-      unsetReadonly(uischema);
+      unsetReadonly(newUiSchema);
     } else {
-      setReadonly(uischema);
+      setReadonly(newUiSchema);
     }
     this.readonly = !this.readonly;
-    this.jsonformService.updateCore(Actions.setUISchema(uischema));
+    this.selectedExample.uischema = newUiSchema;
```

Comment:
This could be replaced with the new readonly support we have in the core, right?

Reply:
As it's only an example app this is not strictly necessary. However maybe it makes sense to also expose the readOnly setter in the forms service for users. In this case we should also adjust the example.

Reply:
I add it to the new root component and thus also to the service